File: Applications/Poisson/MuiltipleEquations/pdeapp1.jl
# create pde structure and mesh structure
pde[1], mesh[1] = Preprocessing.initializeexasim(version);

# Define PDE model: governing equations, initial solutions, and boundary conditions
pde[1].model = "ModelD";            # ModelC, ModelD, ModelW
pde[1].modelfile = "pdemodel1";    # name of a file defining the PDE model
include(pde[1].modelfile * ".jl"); # include the PDE model file
pde[1].modelnumber = 1;

# Set discretization parameters, physical parameters, and solver parameters
pde[1].porder = 3;                  # polynomial degree
pde[1].physicsparam = [1.0 0.0];    # thermal conductivity and boundary value
pde[1].tau = [1.0];                 # DG stabilization parameter

# Choose computing platform and set number of processors
#pde[1].platform = "gpu";           # choose this option if NVIDIA GPUs are available
pde[1].mpiprocs = 1;                # number of MPI processors

# create a linear mesh for a square domain
mesh[1].p, mesh[1].t = Mesh.SquareMesh(8,8,1); # a mesh of 8 by 8 quadrilaterals
# expressions for disjoint boundaries
mesh[1].boundaryexpr = [p -> (p[2,:] .< 1e-3), p -> (p[1,:] .> 1-1e-3), p -> (p[2,:] .> 1-1e-3), p -> (p[1,:] .< 1e-3)];
mesh[1].boundarycondition = [1 1 1 1]; # Set boundary condition for each disjoint boundary

File: Applications/Poisson/MuiltipleEquations/pdeapp1.py
# Create pde object and mesh object
pde[0],mesh[0] = Preprocessing.initializeexasim(version);

# Define a PDE model: governing equations and boundary conditions
pde[0]['model'] = "ModelD";       # ModelC, ModelD, ModelW
pde[0]['modelfile'] = "pdemodel1"; # name of a file defining the PDE model
pde[0]['modelnumber'] = 1;

# Choose computing platform and set number of processors
#pde[0]['platform'] = "gpu";   # choose this option if NVIDIA GPUs are available
pde[0]['mpiprocs'] = 1;        # number of MPI processors

# Set discretization parameters, physical parameters, and solver parameters
pde[0]['porder'] = 3;         # polynomial degree
pde[0]['physicsparam'] = numpy.array([1.0]);   # unit thermal conductivity
pde[0]['tau'] = numpy.array([1.0]);            # DG stabilization parameter

# create a mesh of 8 by 8 quads on a square domain
mesh[0]['p'], mesh[0]['t'] = Mesh.SquareMesh(8,8,1)[0:2];
# expressions for domain boundaries
mesh[0]['boundaryexpr'] = [lambda p: (p[1,:] < 1e-3), lambda p: (p[0,:] > 1-1e-3), lambda p: (p[1,:] > 1-1e-3), lambda p: (p[0,:] < 1e-3)];
mesh[0]['boundarycondition'] = numpy.array([1, 1, 1, 1]); # Set boundary condition for each boundary





